common/src/main/java/com/wynntils/models/beacons/BeaconModel.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.beacons;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.mc.event.AddEntityEvent;
+import com.wynntils.mc.event.RemoveEntitiesEvent;
+import com.wynntils.mc.event.TeleportEntityEvent;
+import com.wynntils.models.beacons.event.BeaconEvent;
+import com.wynntils.models.beacons.type.BeaconColor;
+import com.wynntils.models.beacons.type.UnverifiedBeacon;
+import com.wynntils.models.beacons.type.VerifiedBeacon;
+import com.wynntils.utils.type.TimedSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import net.minecraft.core.Position;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class BeaconModel extends Model {
+    // Amount of armor stands above each other to consider this a beacon
+    private static final int VERIFICATION_ENTITY_COUNT = 6;
+
+    private final TimedSet<UnverifiedBeacon> unverifiedBeacons = new TimedSet<>(1000, TimeUnit.MILLISECONDS, true);
+
+    private final Set<VerifiedBeacon> verifiedBeacons = new HashSet<>();
+
+    public BeaconModel() {
+        super(List.of());
+    }
+
+    @SubscribeEvent
+    public void onEntityAdd(AddEntityEvent event) {
+        if (event.getType() != EntityType.ARMOR_STAND) return;
+
+        Entity entity = event.getEntity();
+        Position position = entity.position();
+
+        if (isDuplicateBeacon(position)) return;
+
+        UnverifiedBeacon unverifiedBeacon = getUnverifiedBeaconAt(position);
+        if (unverifiedBeacon == null) {
+            unverifiedBeacons.put(new UnverifiedBeacon(position, entity));
+            return;
+        }
+
+        boolean correctPosition = unverifiedBeacon.addEntity(entity);
+
+        if (!correctPosition) {
+            unverifiedBeacons.remove(unverifiedBeacon);
+            return;
+        }
+
+        if (unverifiedBeacon.getEntities().size() == VERIFICATION_ENTITY_COUNT) {
+            BeaconColor beaconColor = BeaconColor.fromUnverifiedBeacon(unverifiedBeacon);
```

Comment:
It could be that you are just a bit lucky that this works. Donkey introduced a tick delay in ShamanTotemModel:

```
                    // These must be ran with a delay, as inventory contents are set a couple ticks after the totem
                    // actually spawns
```

Reply:
I don't think so. They are always built up in order, and we also allow a delay. 1000 ms. See timed set.